test/src/test/java/org/corfudb/integration/LogReplicationAbstractIT.java

```diff
@@ -323,6 +293,169 @@ private void triggerSnapshotAndTestRemainingEntries() throws Exception{
+            startLogReplicatorServers();
+
+            // Verify Standby Status Listener received all expected updates (is_data_consistent)
+            log.info(">> Wait ... Replication status UPDATE ...");
+            statusUpdateLatch.await();
+
+            verifyDataOnStandby(numWrites);
+
+            // Confirm replication status reflects snapshot sync completed
+            log.info(">> Verify replication status completed on active");
+            verifyReplicationStatusFromActive();
+
+            verifySnapshotSync(standbyListener, totalStandbyStatusUpdates);
+
+            //reset the listener
+            standbyListener.reset(new CountDownLatch(totalStandbyStatusUpdates));
+
+
+            LogReplicationMetadata.ReplicationEventKey key = LogReplicationMetadata.ReplicationEventKey
+                    .newBuilder().setKey(UUID.randomUUID().toString()).build();
+            LogReplicationMetadata.ReplicationEvent event = LogReplicationMetadata.ReplicationEvent.newBuilder()
+                    .setClusterId(DefaultClusterConfig.getStandbyClusterId())
+                    .setEventId(UUID.randomUUID().toString())
+                    .setType(LogReplicationMetadata.ReplicationEvent.ReplicationEventType.FORCE_SNAPSHOT_SYNC)
+                    .build();
+
+            // add a valid snapshot sync event in the event table
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                txn.putRecord(replicationEventTable, key, event, null);
+                txn.commit();
+            }
+
+            standbyListener.countDownLatch.await();
+            verifySnapshotSync(standbyListener, totalStandbyStatusUpdates);
+
+            while(replicationEventTable.count() != 0) {
+                //wait
+            }
+            // verify the event table is cleared after an event is processed
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                assertThat(txn.getTable(REPLICATION_EVENT_TABLE_NAME).count()).isZero();
+            }
+
+            //reset the listener
+            standbyListener.reset(new CountDownLatch(totalStandbyStatusUpdates));
+
+            // Add an invalid entry to the table so the event listener's onError() gets called in a loop, giving the test to CP/trim
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                txn.putRecord(replicationEventTable, key, LogReplicationMetadata.ReplicationEvent.newBuilder().build(), null);
+                txn.commit();
+            }
+
+            // delete the invalid entry, so the full sync upon resubscription doesn't see the invalid record
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                txn.delete(replicationEventTable, key);
+                txn.commit();
+            }
+
+            // run CP twice so the listener hits a trimmed exception
```

Comment:
the trimmed exception will be hit only if there is new data(onNext()) or resubscription.  If resubscription has already finished by the time of trim, no exception will be hit, right?

Reply:
Yes...so the invalid Entry makes sure that the listener keeps trying to read that entry, fails, resubscribes back to the same address, and again tries to read that invalid entry.

So the listerner is stuck in that loop...and then we delete that invalid entry and run CP twice so that the invalid entry is effectively deleted. Since listener is still stuck in the above loop, it hits a trimmed exception

Reply:
yes.. its related to my question in the first comment.  I am unable to follow where the empty event causes a failure in the listener